qt/scientific_interfaces/test/EnggDiffMultiRunFittingWidgetPresenterTest.h

```diff
@@ -0,0 +1,99 @@
+#ifndef MANTIDQT_CUSTOMINTERFACES_ENGGDIFFMULTIRUNFITTINGWIDGETPRESENTERTEST_H_
+#define MANTIDQT_CUSTOMINTERFACES_ENGGDIFFMULTIRUNFITTINGWIDGETPRESENTERTEST_H_
+
+#include "../EnggDiffraction/EnggDiffMultiRunFittingWidgetPresenter.h"
+#include "EnggDiffMultiRunFittingWidgetModelMock.h"
+#include "EnggDiffMultiRunFittingWidgetViewMock.h"
+
+#include "MantidAPI/WorkspaceFactory.h"
+
+#include <cxxtest/TestSuite.h>
+
+using namespace Mantid;
+
+using namespace MantidQt::CustomInterfaces;
+using testing::Return;
+
+class EnggDiffMultiRunFittingWidgetPresenterTest : public CxxTest::TestSuite {
+public:
+  void test_addFittedPeaks() {
+    auto presenter = setUpPresenter();
+    const API::MatrixWorkspace_sptr ws =
+        API::WorkspaceFactory::Instance().create("Workspace2D", 1, 1, 1);
+
+    EXPECT_CALL(*m_mockModelPtr, addFittedPeaks(123, 1, ws)).Times(1);
+
+    presenter->addFittedPeaks(123, 1, ws);
+    assertMocksUsedCorrectly();
+  }
+
+  void test_addFocusedRun() {
+    auto presenter = setUpPresenter();
+    const API::MatrixWorkspace_sptr ws =
+        API::WorkspaceFactory::Instance().create("Workspace2D", 1, 1, 1);
+
+    EXPECT_CALL(*m_mockModelPtr, addFocusedRun(123, 1, ws)).Times(1);
+
+    const std::vector<std::pair<int, size_t>> workspaceLabels(
+        {std::make_pair(123, 1)});
+    EXPECT_CALL(*m_mockModelPtr, getAllWorkspaceLabels())
+        .Times(1)
+        .WillOnce(Return(workspaceLabels));
+    EXPECT_CALL(*m_mockViewPtr, updateRunList(workspaceLabels));
+
+    presenter->addFocusedRun(123, 1, ws);
+    assertMocksUsedCorrectly();
```

Comment:
Did you consider making two separate tests for this - `test_run_is_added_to_model` and `test_presenter_updates_view`? You can use the `ON_CALL` macro to set return values without asserting that the call was made.